### PR TITLE
model: implement UnmarshalJSON for CIStr

### PR DESCRIPTION
**Pull request by tiancaiamao**
## What have you changed? (mandatory)

So it can be unmarshaled from a single string

## What is the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

unit test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
no 

## Does this PR affect tidb-ansible update? (mandatory)
no

## Does this PR need to be added to the release notes? (mandatory)
no

## Refer to a related PR or issue link (optional)
no
## Benchmark result if necessary (optional)
I'll add one
## Add a few positive/negative examples (optional)

Upgrade from 2.0 to master may meets some problems if  partition tables are used.
Because master use 

```
// PartitionDefinition defines a single partition.
type PartitionDefinition struct {
	ID       int64    `json:"id"`
	Name     CIStr    `json:"name"`
	LessThan []string `json:"less_than"`
	Comment  string   `json:"comment,omitempty"`
}
```

while 2.0 use 
```
// PartitionDefinition defines a single partition.
type PartitionDefinition struct {
	ID       int64    `json:"id"`
	Name     string    `json:"name"`
	LessThan []string `json:"less_than"`
	Comment  string   `json:"comment,omitempty"`
}
```

After this PR, the upgrade would be smooth.

@winkyao @ciscoxll @shenli

### CODE REVIEW

File: model/model.go [hunk 1]
```diff
@@ -425,6 +427,28 @@ func NewCIStr(s string) (cs CIStr) {
...
+		O string `json:"O"` // Original string.
+		L string `json:"L"` // Lower case string.
+	}{}
+	if err := json.Unmarshal(b, &tmp); err == nil {
+		cis.O = tmp.O
+		cis.L = tmp.L
+		return nil
+	}
+
+	// Unmarshal CIStr from a single string.
```
Comment (winkyao) [inline model/model.go]: Add some comment, explain why we need to unmarshal from a single string.

**Comment by ciscoxll** [comment]
LGTM

**Pull request merged**